parlai/tasks/ccpe/agents.py

```diff
@@ -0,0 +1,154 @@
+                    entry['text'] = utterance['text']
+                    currSegments.append(
+                        utterance['segments']
+                    ) if 'segments' in utterance else currSegments.append([])
+                    entry['segments'] = currSegments
+                    currEp.append(entry)
+                    entry = {}
+                    currSegments = []
+
+            flattenedData.append(currEp)
+
+        userData = []
+        assistantData = []
+
+        for ep in range(len(flattenedData)):
+            currUserEp = []
+            currAssistantEp = []
+
+            userCnt = 0
+            asssistantCnt = 0
+            for i, currUtt in enumerate(flattenedData[ep]):
+                if i > 0:
+                    if (
+                        currUtt['speaker'] == 'USER'
+                        and flattenedData[ep][i - 1]['speaker'] == 'ASSISTANT'
+                    ):
+                        entry = []
+                        entry.append(userCnt)
+                        entry.append(currUtt['text'])
+                        entry.append([flattenedData[ep][i - 1]['text']])
+                        entry.append(currUtt['segments'])
+                        entry.append(flattenedData[ep][i - 1]['segments'])
+                        entry.append(False)
+                        currUserEp.append(entry)
+                        userCnt += 1
+                    if (
+                        currUtt['speaker'] == 'ASSISTANT'
+                        and flattenedData[ep][i - 1]['speaker'] == 'USER'
+                    ):
+                        entry = []
+                        entry.append(asssistantCnt)
+                        entry.append(currUtt['text'])
+                        entry.append([flattenedData[ep][i - 1]['text']])
+                        entry.append(currUtt['segments'])
+                        entry.append(flattenedData[ep][i - 1]['segments'])
+                        entry.append(False)
+                        currAssistantEp.append(entry)
+                        asssistantCnt += 1
+
+            currUserEp[-1][5] = True
+            currAssistantEp[-1][5] = True
+            userData.append(currUserEp)
+            assistantData.append(currAssistantEp)
+
+        self.data = assistantData + userData
+
+    def get(self, episode_idx, entry_idx=0):
+        ep = self.data[episode_idx]
+        entry = ep[entry_idx]
+        action = {
+            'id': entry[0],
+            'text': entry[1],
+            'labels': entry[2],
+            'textSegments': entry[3],
+            'labelSegments': entry[4],
+            'episode_done': entry[5],
+        }
+        return action
+
+    def share(self):
+        shared = super().share()
+        shared['data'] = self.data
+        return shared
+
+
+class CCPEAssistantTeacher(CCPEAllTeacher):
+    def _setup_data(self):
+        super()._setup_data()
+        self.data = self.data[:502]
```

Comment:
i think i would prefer having attributes `self.assistantData` and `self.userData`, and then setting `self.data = self.assistantData` here.; either that, or maybe mention what this `502` index represents

Reply:
I'll add `self.assistantData` and `self.userData` !